packages/devtools_app/lib/src/screens/vm_developer/vm_object_model.dart

```diff
@@ -0,0 +1,149 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../service/vm_service_wrapper.dart';
+import '../../shared/globals.dart';
+import '../debugger/debugger_model.dart';
+
+/// Wrapper class for storing Dart VM objects with their relevant VM
+/// information.
+abstract class VmObject {
+  VmObject({required this.ref});
+
+  final ObjRef ref;
+
+  String? get name;
+
+  VmServiceWrapper get _service => serviceManager.service!;
+
+  IsolateRef? _isolate;
+
+  Obj? _obj;
+
+  Future<void> fetchObject() async {
+    final isolateRef = serviceManager.isolateManager.selectedIsolate.value!;
+    _isolate = await _service.getIsolate(isolateRef.id!);
+    _obj = await _service.getObject(_isolate!.id!, ref.id!);
+  }
+
+  Future<void> initialize();
+}
+
+class ClassObject extends VmObject {
+  ClassObject({required super.ref});
+
+  Class? get obj => _clazz;
+  Class? _clazz;
+
+  @override
+  String? get name => obj?.name;
+
+  Script? get script => _script;
+  Script? _script;
+
+  SourcePosition? get pos => _pos;
+  SourcePosition? _pos;
+
+  InstanceSet? get instances => _instances;
+  InstanceSet? _instances;
+
+  @override
+  Future<void> initialize() async {
+    await fetchObject();
+
+    _clazz = _obj as Class?;
+
+    if (_clazz != null) {
+      final token = _clazz!.location!.tokenPos!;
+      _script = await _service.getObject(
+        _isolate!.id!,
+        _clazz!.location!.script!.id!,
+      ) as Script;
+      _pos = SourcePosition.calculatePosition(
+        _script!,
+        token,
+      );
+    }
```

Comment:
as we implement the object inspector for other Vm object types, we should see if any of this can be generalized if it applies to other data types as well.